gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+// Implementations of netCDF functions used for 
+// implementing CF-1.8 convention awareness
+//
+// Author: wchen329
+#include <cstdio>
+#include <cstring>
+#include <vector>
+#include "netcdf.h"
+#include "netcdfdataset.h"
+#include "netcdfsg.h"
+namespace nccfdriver
+{
+	/* Re-implementation of mempcpy
+	 * but compatible with libraries which only implement memcpy
+	 */
+	static void* memcpy_jump(void *dest, const void *src, size_t n)
+	{
+		memcpy(dest, src, n);
+		char * a = (char*)dest + n; // otherwise unneccesary casting to stop compiler warnings
+		void * b = (void*)a;	
+		return b;
+	}
+
+	/* Attribute Fetch
+	 * -
+	 * A function which makes it a bit easier to fetch single text attribute values
+	 * attrgets returns a char. seq. on success, nullptr on failure 
+	 * attrgets takes in ncid and varId in which to look for the attribute with name attrName 
+	 *
+	 * The returned char. seq. must be free'd afterwards
+	 */
+	char* attrf(int ncid, int varId, const char * attrName)
+	{
+		size_t len = 0;
+		nc_inq_attlen(ncid, varId, attrName, &len);
+		
+		if(len < 1)
+		{
+			return nullptr;
+		}
+
+		char * attr_vals = new char[len];
+		// Now look through this variable for the attribute
+		if(nc_get_att_text(ncid, varId, attrName, attr_vals) != NC_NOERR)
+		{
+			delete[] attr_vals;
+			return nullptr;		
+		}
+
+		return attr_vals;
+	}
+
+	/* Point
+	 * (implementations)
+	 *
+	 */
+	Point::~Point()
+	{
+		delete[] this->values;
+	}
+
+	/* SGeometry 
+	 * (implementations)
+	 *
+	 */
+	SGeometry::SGeometry(int ncId, int geoVarId)
+		: gc_varId(geoVarId), touple_order(0), current_vert_ind(0), cur_geometry_ind(0), cur_part_ind(0)
+	{
+
+		char * cart = nullptr;
+		this->pt_buffer = nullptr;
+		memset(this->container_name, 0, NC_MAX_NAME+1);
+
+		// Get geometry container name
+		if(nc_inq_varname(ncId, geoVarId, container_name) != NC_NOERR)
+		{
+			throw new SG_Exception_Existential((char*)"new geometry container", "the variable of the given ID"); 	
```

Comment:
cast to remove once constructor signature is fixed

Reply:
:+1: will fix